src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -81,6 +84,21 @@ public void updatePerson(ReadOnlyPerson target, ReadOnlyPerson editedPerson)
         indicateAddressBookChanged();
     }
 
+    @Override
+    public void changeTag(Tag oldTag, Tag newTag) throws PersonNotFoundException, DuplicatePersonException {
+        for (int i = 0; i < addressBook.getPersonList().size(); i++) {
+            ReadOnlyPerson oldPerson = addressBook.getPersonList().get(i);
+
+            Person newPerson = new Person(oldPerson);
+            Set<Tag> newTags = newPerson.getTags();
+            newTags.remove(oldTag);
+            newTags.add(newTag);
+            newPerson.setTags(newTags);
```

Comment:
hmm I don't think this line is necessary, you can double check if you wish to make your code more compact